src/v/archival/service.h

```diff
@@ -183,7 +134,7 @@ class scheduler_service_impl {
     ss::gate _gate;
     ss::abort_source _as;
     ss::semaphore _stop_limit;
-    ntp_upload_queue _queue;
+    absl::flat_hash_map<model::ntp, ss::lw_shared_ptr<ntp_archiver>> _archivers;
```

Comment:
should we consider using container without continuous allocation ? I think there may be thousands of partitions

Reply:
I've changed it to `btree_map`, but the downside is that ntp comparison may be costly.